wlm/src/main/java/ai/djl/serving/wlm/LmiUtils.java

```diff
@@ -45,67 +45,98 @@ public final class LmiUtils {
+                    Map.entry("aquila", "vllm"),
+                    Map.entry("baichuan", "vllm"),
+                    Map.entry("bloom", "vllm"),
+                    Map.entry("chatglm", "vllm"),
+                    Map.entry("deci", "vllm"),
+                    Map.entry("gemma", "vllm"),
+                    Map.entry("gpt2", "vllm"),
+                    Map.entry("gptj", "vllm"),
+                    Map.entry("internlm2", "vllm"),
+                    Map.entry("mistral", "vllm"),
+                    Map.entry("mixtral", "vllm"),
+                    Map.entry("opt", "vllm"),
+                    Map.entry("phi", "vllm"),
+                    Map.entry("qwen", "vllm"),
+                    Map.entry("qwen2", "vllm"),
+                    Map.entry("stablelm", "vllm"));
 
     private LmiUtils() {}
 
     static String inferLmiEngine(ModelInfo<?, ?> modelInfo) throws ModelException {
-        // MMS/Torchserve
-        if (Files.isDirectory(modelInfo.modelDir.resolve("MAR-INF"))) {
-            logger.info("Found legacy torchserve model, use Python engine.");
-            return "Python";
-        }
-
-        Properties prop = modelInfo.prop;
+        Properties prop = modelInfo.getProperties();
         HuggingFaceModelConfig modelConfig = getHuggingFaceModelConfig(modelInfo);
+        String features = Utils.getEnvOrSystemProperty("SERVING_FEATURES");
+        String rollingBatch = prop.getProperty("option.rolling_batch", "auto");
         if (modelConfig == null) {
-            String engineName = isTrtLLM(modelInfo) ? "MPI" : "Python";
+            String engineName = isTrtLLM(rollingBatch, features) ? "MPI" : "Python";
             logger.info("No config.json found, use {} engine.", engineName);
             return engineName;
         }
-        String features = Utils.getenv("SERVING_FEATURES");
+
         String modelType = modelConfig.getModelType();
-        String engineName;
+        String engineName = "Python";
         if ("stable-diffusion".equals(modelType)) {
             // TODO: Move this from hardcoded to deduced in PyModel
             prop.setProperty("option.entryPoint", "djl_python.stable-diffusion");
-            engineName = "DeepSpeed";
-        } else if (isDeepSpeedRecommended(modelType)) {
-            engineName = "DeepSpeed";
-        } else if (features != null && features.contains("trtllm")) {
             engineName = "MPI";
-        } else {
+        } else if (isTrtLLM(rollingBatch, features)) {
+            engineName = "MPI";
+        } else if ("auto".equals(rollingBatch)) {
+            rollingBatch = inferRollingBatchForModel(modelType, features);
+        }
+
+        if ("lmi-dist".equals(rollingBatch)) {
+            engineName = "MPI";
+        } else if ("vllm".equals(rollingBatch)) {
             engineName = "Python";
+            // workaround for vllm TP
+            String tpDegree = prop.getProperty("option.tensor_parallel_degree");
```

Comment:
option. field should take higher priority than environment variable. Customer could also potentially set `OPTION_TENSOR_PARALLEL_DEGREE` did the ENV parsing happening before this logic @frankfliu ?

Reply:
yea i have this fix ready in local - I have moved around some things as well to make it a bit cleaner. It will still live in LMIUtils for now, but should be easier to separate into a plugin after this